frontend/src/metabase/hooks/use-loading-timer.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react";
+
+interface loadingTimerProps {
+  timer: number;
+  onTimeout: () => void;
+}
+
+export function useLoadingTimer(isLoading: any, props: loadingTimerProps) {
+  const [trigger, setTrigger] = useState(false);
+  const { onTimeout, timer } = props;
+
+  useEffect(() => {
+    if (isLoading) {
+      const timeoutId = setTimeout(() => {
+        if (isLoading) {
+          setTrigger(true);
+          if (onTimeout) {
+            onTimeout();
+          }
```

Comment:
Why check if `onTimeout` is not defined if it's declared as mandatory in `LoadingTimerProps`?

Reply:
Haha, I guess that's the magic of types and not just vanilla JS. I can actually depend on things being there